htdocs/comm/propal/card.php

```diff
@@ -823,8 +823,15 @@
 				$type = $prod->type;
 				$fk_unit = $prod->fk_unit;
 			} else {
-				$pu_ht = price2num($price_ht, 'MU');
-				$pu_ttc = price2num(GETPOST('price_ttc'), 'MU');
+				if (!empty($conf->multicurrency->enabled)) {// we have to preserve all precision for the conversion as editline
```

Comment:
Your example and solution work because you choose 100$ and there is no decimal to it. But imagine user wants to sell with unit price of 100,12345$ and usd rate is 1.273456 (converstion rates can have 6 decimals)
Then you will still have problem. (disabling a rounding is not disabling a rounding, just increase number of decimal to max allowed by database, currently 8, instead of max defined by user). And 8 may not be enough in a more complex case.

What you need for unit price accuracy for amount in database is :  
accuracy of the foreign currency (usually the 5 we have in setup) + accuracy introduced by increase of decimals due to rate (so 6) = 11 instead of 5. And currently fields into database have accuracy of 8 maximum.

So to fix this, we must:
- increase fields of all unit amount to 16 decimals instead of 8 ( double(32,16) instead of double(24,8))
- have a mecanism so setup of "unit price accuracy" is increase by 6 when module multicurrency is enabled, required to get price in foreign currency to be value of setup.